### PR TITLE
redo(ticdc): fix a bug about redo manager error handling

### LINKED ISSUE #6764: cdc redo manager doesn't handle internal errors correctly
### What did you do?

Deploy a cluster and inject some errors into redo manager.

### What did you expect to see?

Errors can be handled correctly.

### What did you see instead?

Errors are ignored incorrectly.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #6764 

### What is changed and how it works?

Handling errors correctly in redo manager. The bug is introduced in #6671 .

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Review by hi-rustin — COMMENTED** [review]
Could you add a test for `NewOwnerManagerOptions` to make sure we set up the right values?
I think it will help us to avoid this happening again.
This is also helpful to make sure we set up the right `ture` or `false`.

File: cdc/redo/manager.go [hunk 1]
```diff
@@ -660,3 +672,12 @@ func (m *ManagerImpl) bgGC(ctx context.Context) {
 		}
 	}
 }
+
+func (m *ManagerImpl) withLock(action func(m *ManagerImpl) error) error {
+	m.rwlock.RLock()
+	defer m.rwlock.RUnlock()
+	if atomic.LoadInt32(&m.closed) != 0 {
+		return cerror.WrapError(cerror.ErrRedoWriterStopped, errors.New("redo manager is stopped"))
```
Comment (hi-rustin) [inline cdc/redo/manager.go:680]: Can we use cerror.ErrRedoWriterStopped.genStackxxx？

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

File: cdc/redo/options_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	opts := NewProcessorManagerOptions(ch)
+	require.True(t, opts.EnableBgRunner)
+	require.True(t, opts.EnableGCRunner)
+	require.False(t, opts.EmitMeta)
+	require.True(t, opts.EmitRowEvents)
+	require.False(t, opts.EmitDDLEvents)
+	select {
+	case opts.ErrCh <- nil:
+	default:
+		panic("ch shouldn't be full")
```
Comment (hi-rustin) [inline cdc/redo/options_test.go:48]:
```suggestion
		t.Fatal("ch shouldn't be full")
```

**Pull request merged**